src/ocamlorg_web/lib/handler.ml

```diff
@@ -649,12 +670,31 @@ let packages_autocomplete_fragment t req =
         Ocamlorg_package.search ~is_author_match ~sort_by_popularity:true t
           search
       in
-      let results = List.map (Package_helper.frontend_package t) packages in
-      let top_5 = results |> List.take 5 in
+
+      let open Lwt.Syntax in
+      let documentation_status (pkg : Ocamlorg_package.t) =
+        let* package_documentation_status =
+          Ocamlorg_package.documentation_status ~kind:`Package pkg
+        in
+        Lwt.return
+          (match package_documentation_status with
+          | Some { failed = false; _ } -> Ocamlorg_frontend.Package.Success
+          | Some { failed = true; _ } -> Failure
+          | None -> Unknown)
+      in
+
+      let* top_5 =
+        packages |> List.take 5
+        |> Lwt_list.map_p (fun pkg ->
+               let+ doc_status = documentation_status pkg in
+               (Package_helper.frontend_package t pkg, doc_status))
+      in
```

Comment:
This and the version in the *packages_search* function above are almost identical. I want to remove that duplication but don't know where's the best place (module/file) to put the common logic.

Reply:
There's a module `Package_helper` inside `handler.ml` that holds some similar logic of translating between the `Ocamlorg_package` notion of what a package is and what a package looks to `Ocamlorg_frontend.Package`.